firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import ConfigParser
+import os
+import re
+
+import mozinfo
+
+from ..base import BaseLib
+from .appinfo import AppInfo
+from .prefs import Preferences
+
+
+class SoftwareUpdate(BaseLib):
+    """The SoftwareUpdate API adds support for an easy access to the update process."""
+
+    PREF_APP_DISTRIBUTION = 'distribution.id'
+    PREF_APP_DISTRIBUTION_VERSION = 'distribution.version'
+    PREF_APP_UPDATE_URL = 'app.update.url'
+    PREF_DISABLED_ADDONS = 'extensions.disabledAddons'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.app_info = AppInfo(marionette_getter)
+        self.prefs = Preferences(marionette_getter)
+
+        self.update_channel = UpdateChannel(self)
+        self.mar_channels = MARChannels(self)
+        self.active_update = ActiveUpdate(self)
+
+        self._aus = self.marionette.execute_script("""
+            return Cc['@mozilla.org/updates/update-service;1']
+                .getService(Ci.nsIApplicationUpdateService);
+            """)
```

Comment:
Lets not return the service here. It will be JSON string but no real service anymore, and you won't be able to convert it back. We can access properties from it but no methods. So lets simply add this code to all the methods where it is used.